server/testutil/testregistry/testregistry.go

```diff
@@ -99,11 +116,23 @@ func (r *Registry) ImageAddress(imageName string) string {
 	return fmt.Sprintf("%s:%d/%s", r.host, r.port, imageName)
 }
 
+func (r *Registry) remoteOpts() []remote.Option {
+	if r.auth == nil {
+		return nil
+	}
+	return []remote.Option{
+		remote.WithAuth(&authn.Basic{
+			Username: r.auth.Username,
+			Password: r.auth.Password,
+		}),
+	}
+}
+
 func (r *Registry) Push(t *testing.T, image gcr.Image, imageName string) string {
 	fullImageName := r.ImageAddress(imageName)
 	ref, err := name.ParseReference(fullImageName)
 	require.NoError(t, err)
-	err = remote.Write(ref, image)
+	err = remote.Write(ref, image, r.remoteOpts()...)
```

Comment:
should the caller have to pass in these creds? (kind of looks like the server is self-authorizing the request? or am I misunderstanding)